File: common/src/main/java/generations/gg/generations/core/generationscore/world/item/LangTooltip.java
package generations.gg.generations.core.generationscore.world.item;

import net.minecraft.locale.Language;
import net.minecraft.network.chat.Component;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.TooltipFlag;
import net.minecraft.world.level.Level;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface LangTooltip {

    default void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> tooltipComponents, TooltipFlag isAdvanced) {
        var tooltipId = tooltipId(stack);
        if(Language.getInstance().has(self().getDescriptionId(stack))) tooltipComponents.add(Component.translatable(tooltipId));
    }

    default String tooltipId() {
        return tooltipId(null);
    }

    private String tooltipId(ItemStack stack) {
        return self().getDescriptionId(stack) + ".tooltip";
    }

    private Item self() {
        return (Item) this;
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/item/LangTooltip.kt
package generations.gg.generations.core.generationscore.common.world.item

import net.minecraft.locale.Language
import net.minecraft.network.chat.Component
import net.minecraft.world.item.Item
import net.minecraft.world.item.Item.TooltipContext
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.TooltipFlag

interface LangTooltip {
    public fun addText(
        stack: ItemStack,
        level: TooltipContext,
        tooltipComponents: MutableList<Component>,
        isAdvanced: TooltipFlag
    ) {
        val tooltipId = tooltipId(stack)
        if (Language.getInstance().has(self().getDescriptionId(stack))) tooltipComponents.add(
            Component.translatable(
                tooltipId
            )
        )
    }

    fun tooltipId(): String {
        return tooltipId(null)
    }

    fun tooltipId(stack: ItemStack?): String {
        return self().getDescriptionId(stack ?: ItemStack.EMPTY) + ".tooltip"
    }

    private fun self(): Item {
        return this as Item
    }
}
